components/post_view/post_list_virtualized/post_list_virtualized.jsx

```diff
@@ -354,19 +435,56 @@ class PostList extends React.PureComponent {
         return offsetFromBottom <= BUFFER_TO_BE_CONSIDERED_BOTTOM && scrollHeight > 0;
     }
 
+    hideNewMessagesToast = (dismiss = true) => {
+        if (this.state.showNewMessagesToast) {
+            if (dismiss) {
+                this.setState({
+                    showNewMessagesToast: false,
+                    lastViewedBottom: Date.now()
+                });
+                return;
+            }
+            this.hideNewMessagesTimeout = setTimeout(() => {
```

Comment:
Could the Toast component itself be in charge of fading itself out? The react-bootstrap components do that by having an internal `show` state that is changed based on the `show` prop provided by the caller.

Reply:
also, if there needs to be a different behaviour depending on which toast it is, you could probably pass that on the construction and simply having to call `hide`, letting the component handle that.